packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebasePermissionManager.java

```diff
@@ -24,15 +24,16 @@ interface RequestPermissionsSuccessCallback {
   @Override
   public boolean onRequestPermissionsResult(
       int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    requestInProgress = false;
-    if (requestCode != permissionCode) {
+    if (requestInProgress && requestCode == permissionCode || this.successCallback != null) {
```

Comment:
shouldn't we use **and** for successCallback check?
```
if (requestInProgress && requestCode == permissionCode || this.successCallback != null)
```

I see that that [here](https://github.com/firebase/flutterfire/blob/f48b365112b342a373c0f7655fc27ef4dbd1bd9a/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebasePermissionManager.java#L55) we set `successCallback` before we set [requestInProgress = true](https://github.com/firebase/flutterfire/blob/f48b365112b342a373c0f7655fc27ef4dbd1bd9a/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebasePermissionManager.java#L62), but is it safe enough?

Reply:
Yes, correct, that should be `&&`. Good catch! 👍

Reply:
It is safe, every time we call `requestPermission()`, we set the callback.